src/com/twu/biblioteca/MainMenu.java

```diff
@@ -32,36 +34,71 @@ public void showGoodbyeMessage() {
         System.out.println(goodBye);
     }
 
+    public void showBookCheckOutReferenceMessage() {
+        System.out.println(checkOutBookReference);
+    }
+
     public void showOptions() {
-        for(int i = 0; i < options.size(); i++) {
+        for (int i = 0; i < options.size(); i++) {
             System.out.println(options.get(i));
         }
         System.out.print(">");
     }
 
-    public void showList() {
-        for(int i = 0; i < books.size(); i++) {
-            System.out.println(books.get(i));
-        }
-    }
-
     public void manageOptions() {
         int counter = 0;
-        Scanner scanner = new Scanner(System.in);
+        Scanner optionsScanner = new Scanner(System.in);
 
         while (counter == 0) {
-            String userInput = scanner.next();
-            if(userInput.equals("1")) {
-                showList();
-                counter = 1;
-            }
-            else if(userInput.equals("2")) {
+            String userInput = optionsScanner.next();
+            if (userInput.equals("1")) {
+                showBookList();
+                showOptions();
+            } else if (userInput.equals("2")) {
+                showBookList();
+                manageBookList();
+            } else if (userInput.equals("3")) {
                 showGoodbyeMessage();
                 System.exit(0);
-            }
-            else {
+            } else {
                 showInvalidMessage();
             }
         }
     }
+
+    public void showBookList() {
+        for (int i = 0; i < this.books.size(); i++) {
+            Book book = this.books.get(i);
+            if (book.getCheckedOut() == false) {
+                int count = countCheckedOutBooks();
+                System.out.println(String.format(
+                        "%d. %s | %s | %d",
+                        (i + 1) - count, book.getTitle(), book.getAuthor(), book.getYearPublished()));
+            }
+        }
+    }
+
+    public int countCheckedOutBooks() {
+        int count = 0;
+        for (Book b : this.books){
+            if (b.getCheckedOut()) count++;
+        }
+        return count;
+    }
```

Comment:
Java also (now) has a bunch of methods that operate on collections to support this sort of processing without having to write an explicit `for` loop.  Many of these methods have similar names to JS and provide a far more functional way to write collection processing code.  If you're feeling confident with your Java skills, it's a nice refactoring to look at.

Reply:
So I could use an iterator?

Reply:
Yep!  Java streams/iterators are now at the point where you can replace almost all explicit iterations with an appropriate use of `map`, `reduce`, `collect`, etc.  The syntax is not as clean as JS, but it's not too bad either.

Alternatively, you could make a good argument for the explicit looping being more easy to read, but - as an industry - I think the average developer is getting better at pulling these tools into their standard toolkit, so I encourage their use.